Source/Actions/Microsoft.Deployment.Actions.Custom/Cuna/GetCunaByosDetails.cs

```
@@ -0,0 +1,104 @@
+                    return new ActionResponse(ActionStatus.Success, response.ToString(), true);
+                }
+                else
+                {
+                    return new ActionResponse(ActionStatus.Failure, response, null, "CunaApiCallFailure", response["message"]?.ToString());
+                }
+            }
+           return new ActionResponse(ActionStatus.Failure, null , null, "CunaApiNoResponse");
+        }
+
+        public async Task<JObject> GetByosDetails(
+            string cunaApiUrl,
+            string userId,
+            string contractId,
+            string customerUpn,
+            string tenantId,
+            string authToken
+        )
+        {
+            var postData = new Dictionary<string, string>
+            {
+                {"userId", userId},
+                {"contractId", contractId},
+                {"customerUpn", customerUpn},
+                {"customerTenant", tenantId }
+            };
+
+            ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12 | SecurityProtocolType.Tls11;
+            using (var client = new HttpClient())
+            {
+                client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", authToken);
+                var content = new StringContent(JsonConvert.SerializeObject(postData), Encoding.UTF8, "application/json");
+
+                var response = await client.PostAsync(cunaApiUrl, content);
+                if (!response.IsSuccessStatusCode)
+                {
+                    throw new Exception("Unable to contact Cuna API service.");
+                }
+                var jsonBody = await response.Content.ReadAsStringAsync();
+                return GetJsonObject(jsonBody);
+            }
+        }
+
+        public JObject GetJsonObject(string jsonBody)
+        {
+            try
+            {
+                var obj = JsonUtility.GetJsonObjectFromJsonString(jsonBody);
+                return obj;
```

I appreciate trying to mask information coming from the server, but we shouldn't just catch exceptions and return null. We should throw the exception with a meaningful message.